mp4parse/src/lib.rs

```diff
@@ -773,6 +947,11 @@ fn read_moov<T: Read>(f: &mut BMFFBox<T>, context: &mut MediaContext) -> Result<
                 debug!("{:?}", pssh);
                 vec_push(&mut context.psshs, pssh)?;
             }
+            BoxType::UserdataBox => {
+                let udta = read_udta(&mut b)?;
```

Comment:
My primary concern with these additions is that the main user of mp4parse-rust (Gecko) is exposed to new forms of parse failure if invalid udta (or children) are encountered.  We don't want to reject MP4s we previously accepted due to content Gecko isn't interested in.

I think a simple solution to this is to make MediaContext::userdata a `Result<Option<Userdata>>` instead.  That does make the MediaContext API slightly odd, though.

Reply:
Instead of `Result<Option<UserdataBox>>`, I elected to make it `Option<Result<UserdataBox>>` instead, since `Result<Option<_>>` doesn't implement `Default`. 

I considered just ignoring the error and returning `None` if udta parsing fails, so as to not introduce any weirdness into the MediaContext API, but that seems to me that it would conflate the lack of an udta atom with a corrupt udta atom.